subsys/net/ip/tcp2.c

```diff
@@ -832,6 +940,45 @@ static struct tcp *tcp_conn_new(struct net_pkt *pkt)
 	return conn;
 }
 
+static void update_acked_data(struct tcp *conn, struct tcphdr *th)
+{
+	if (!th) {
+		return;
+	}
+
+	if (!net_tcp_seq_greater(th_ack(th), conn->seq)) {
+		/* If there is any pending data, remove them
+		 * from re-send as peer just acked it
+		 */
+		u32_t acked_len = th_ack(th) - conn->last_seq;
+		int ret;
+
+		if (conn->pending_data && acked_len) {
+			NET_DBG("Peer acked %u bytes", acked_len);
+
+			net_pkt_cursor_init(conn->pending_data);
+			net_pkt_set_overwrite(conn->pending_data, true);
+
+			ret = net_pkt_pull(conn->pending_data, acked_len);
+			if (ret < 0 && conn->pending_len != (int)acked_len) {
```

Comment:
Would it be a good idea to check error cases with pending_len, acked_len before net_pkt_pull()?

Reply:
Only thing to check is to calculate the length of the pending_data. IMHO this is too costly to do, as the pending_data can be quite large and using net_pkt_get_len() for each incoming ACK is heavy. If we would keep track of pending_data length separately, then it would be much better. I suggest we leave this to future patches.